File: python/p019.py
# Counting Sundays
#
# How many Sundays fell on the first of the month during twentieth century
# (1 Jan 1901 to 31 Dec 2000)

SUNDAY = 6 # integer weekday representation of sunday
total = 0

from datetime import date

for year in range(1901,2001):
    for month in range(1,13):
        if date(year,month,1).weekday() == SUNDAY:
            total += 1

print(total)

# output: 171

File: go/p019.go
package main

import (
	"fmt"
	"time"
)

func main() {
	total := 0
	for year := 1901; year <= 2000; year++ {
		for month := time.January; month <= time.December; month++ {
			date := time.Date(year, month, 1, 0, 0, 0, 0, time.UTC)
			if date.Weekday() == time.Sunday {
				total += 1
			}
		}
	}
	fmt.Println(total)
	// output: 171
}
